node/pkg/watchers/stacks/watcher.go

```diff
@@ -0,0 +1,652 @@
+		Payload:          msgData.Payload,
+		Sequence:         msgData.Sequence,
+		IsReobservation:  isReobservation,
+	}
+
+	// Submit the message to the channel for processing
+	w.msgC <- msgPub
+
+	return nil
+}
+
+/// HELPERS
+
+// Extracts the event name from an event tuple
+func extractEventName(eventTuple *clarity.Tuple) (string, error) {
+	eventNameVal, ok := eventTuple.Values["event"]
+	if !ok {
+		return "", fmt.Errorf("missing 'event' field in tuple")
+	}
+
+	// Check if event is a StringASCII or StringUTF8
+	var eventName string
+	if strVal, ok := eventNameVal.(*clarity.StringASCII); ok {
+		eventName = strVal.Value
+	} else if strVal, ok := eventNameVal.(*clarity.StringUTF8); ok {
+		eventName = strVal.Value
+	} else {
+		return "", fmt.Errorf("'event' field is not a string type: %T", eventNameVal)
+	}
+
+	return eventName, nil
+}
+
+// Extracts core message fields from an event tuple
+func extractMessageData(eventTuple *clarity.Tuple) (*MessageData, error) {
+	// Get the data field which should contain the message
+	dataVal, ok := eventTuple.Values["data"]
+	if !ok {
+		return nil, fmt.Errorf("missing 'data' field in tuple")
+	}
+
+	// Cast data to tuple
+	msgTuple, ok := dataVal.(*clarity.Tuple)
+	if !ok {
+		return nil, fmt.Errorf("'data' field is not a tuple: %T", dataVal)
+	}
+
+	// Extract message fields
+	emitterVal, ok := msgTuple.Values["emitter"]
+	if !ok {
+		return nil, fmt.Errorf("missing 'emitter' field in message")
+	}
+
+	emitterBuffer, ok := emitterVal.(*clarity.Buffer)
+	if !ok || emitterBuffer.Len() != 32 {
+		return nil, fmt.Errorf("'emitter' field is not a 32-byte buffer: %T", emitterVal)
+	}
+
+	// Convert buffer to wormhole address
+	emitterAddr := vaa.Address{}
+	copy(emitterAddr[:], emitterBuffer.Data[:])
+
+	nonceVal, ok := msgTuple.Values["nonce"]
+	if !ok {
+		return nil, fmt.Errorf("missing 'nonce' field in message")
+	}
+
+	nonceUint, ok := nonceVal.(*clarity.UInt128)
+	if !ok || nonceUint.Value.Cmp(maxUint32BigInt) > 0 {
+		return nil, fmt.Errorf("invalid 'nonce' field: %T", nonceVal)
+	}
+
+	sequenceVal, ok := msgTuple.Values["sequence"]
+	if !ok {
+		return nil, fmt.Errorf("missing 'sequence' field in message")
+	}
+
+	sequenceUint, ok := sequenceVal.(*clarity.UInt128)
+	if !ok || sequenceUint.Value.Cmp(maxUint64BigInt) > 0 {
+		return nil, fmt.Errorf("invalid 'sequence' field: %T", sequenceVal)
+	}
+
+	consistencyLevelVal, ok := msgTuple.Values["consistency-level"]
+	if !ok {
+		return nil, fmt.Errorf("missing 'consistency-level' field in message")
+	}
+
+	consistencyLevelUint, ok := consistencyLevelVal.(*clarity.UInt128)
+	if !ok || consistencyLevelUint.Value.Cmp(maxUint8BigInt) > 0 {
+		return nil, fmt.Errorf("invalid 'consistency-level' field: %T", consistencyLevelVal)
+	}
+
+	payloadVal, ok := msgTuple.Values["payload"]
+	if !ok {
+		return nil, fmt.Errorf("missing 'payload' field in message")
+	}
+
+	payload, ok := payloadVal.(*clarity.Buffer)
+	if !ok || payload.Len() > 8192 {
```

Comment:
Could you move the value `8192` into a constant and document that this is related to the Clarity transaction size limit?

It may be helpful to create a `consts.go` where the constants are stored for the Stacks package.